### DISCUSSION TITLE
How to debug the code?

### DISCUSSION THREAD

**Discussion by cunyizju** [OpenRadioss/OpenRadioss#1289] [Q&A]
Hey, debugging may be an excellent way to understand the structure of OpenRadioss. I have installed GDB.
With the following command, I can obtain the executable for debugging. 
```
./build_script.sh -arch=linux64_gf -debug=1 -nt=4
```
However, when I set the breakpoint in 'starter.F' and ran the starter with *0000.rad, the program did not stop at the breakpoint.
Could you please let me know the correct way to debug?

**Comment by laurent-altr** [comment]
Hi @cunyizju, 
There is no known issue with GDB and OpenRadioss. 
Compiling with "debug=1" build a special binary with the suffix "_db": you need to use this one for your debugging session.

Thank you

**Comment by cunyizju** [comment]
Thanks, @laurent-altr. I have obtained the "_db" file. However, how to use the file? I mean, how to set the breakpoint and read the variables when running the program? I want to run the code line by line.

Best wishes,
Bruce

**Comment by laurent-altr** [comment]
Hi, 

For instance you can use the gdb command line. Let's say you want to start from a breakpoint in lectur.F at line 1112 for the model TRITUBV4_0000.rad. 

```
gdb ~/OpenRadioss/exec/starter_linux64_gf_db
(gdb) b lectur.F:1112
(gdb) run -i TRITUBV4_0000.rad
Breakpoint 1, lectur (multi_fvm=..., lsubmodel=..., is_dyna=0, detonators=..., ebcs_tab=..., seatbelt_converted_elements=..., nb_seatbelt_shells=0, nb_dyna_include=0) at ~/OpenRadioss/starter/source/starter/lectur.F:1112
````

(for the engine, you may want to start from a breakpoint into resol.F).

**Comment by cunyizju** [comment]
Many thanks @laurent-altr. Will try it later :)

Best wishes,
Bruce

**Comment by elequiniou** [comment]
Thanks a lot Bruce and Laurent! As the issue has been answered and as it is not really an issue vs the code, I propose to convert it to a discussion to help others having similar questions regarding debug.
